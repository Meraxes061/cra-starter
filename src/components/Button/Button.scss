$primary-color: #FFA000;
$secondary-color: #FF4081;
$text-color: white;

.c-button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    line-height: 1rem;
    border: 1px solid transparent;
    border-radius: 4px;
    background: transparent;
    user-select: none;
    cursor: pointer;

    &--primary {
        background-color: $primary-color;
        color: $text-color
    }

    &--secondary {
        background-color: $secondary-color;
        color: $text-color;
    }

    &--small {
        font-size: .75rem;
        padding: .25em .5em;
        line-height: 0.75rem;
    }

    &--medium {
        font-size: 1rem;
        padding: .5em 1em;
    }

    &--large {
        font-size: 1.25rem;
        padding: 1em 2em;
        line-height: 1.5rem;
    }
}