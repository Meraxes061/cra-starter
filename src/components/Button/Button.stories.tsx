import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';

storiesOf('Button', module).add('Default', () => (
    <div style={{ fontFamily: 'Verdana' }}>
        <h3 style={{ backgroundColor: 'maroon', color: 'white', padding: '.5rem 1rem' }}>Default</h3>
        <div style={{ display: 'flex' }}>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button</h4>
                <Button>Default Button</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button With Loading</h4>
                <Button isLoading>Default Button</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button Small</h4>
                <Button size='small'>Default Button Small</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button Medium (Default)</h4>
                <Button size='medium'>Default Button Medium</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button Large</h4>
                <Button size='large'>Default Button Large</Button>
            </div>
        </div>


    </div>
));

storiesOf('Button', module).add('Primary', () => (
    <div style={{ fontFamily: 'Verdana' }}>

        <h3 style={{ backgroundColor: 'maroon', color: 'white', padding: '.5rem 1rem' }}>Primary</h3>
        <div style={{ display: 'flex' }}>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Primary Button</h4>
                <Button theme='primary'>Primary Button</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Primary Button With Loading</h4>
                <Button theme='primary' isLoading>Primary Button</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button Small</h4>
                <Button theme='primary' size='small'>Primary Button Small</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button Medium (Default)</h4>
                <Button theme='primary' size='medium'>Primary Button Medium</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Default Button Large</h4>
                <Button theme='primary' size='large'>Primary Button Large</Button>
            </div>
        </div>
    </div>
))
storiesOf('Button', module).add('Secondary', () => (
    <div style={{ fontFamily: 'Verdana' }}>

        <h3 style={{ backgroundColor: 'maroon', color: 'white', padding: '.5rem 1rem' }}>Secondary</h3>
        <div style={{ display: 'flex' }}>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Secondary Button</h4>
                <Button theme='secondary'>Secondary Button</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Secondary Button With Loading</h4>
                <Button theme='secondary' isLoading>Secondary Button</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Secondary Button Small</h4>
                <Button theme='secondary' size='small'>Secondary Button Small</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Secondary Button Medium (Default)</h4>
                <Button theme='secondary' size='medium'>Secondary Button Medium</Button>
            </div>
            <div style={{ marginRight: 16, backgroundColor: '#eee', textAlign: 'center', padding: '1rem' }}>
                <h4>Secondary Button Large</h4>
                <Button theme='secondary' size='large'>Secondary Button Large</Button>
            </div>
        </div>
    </div>
))
