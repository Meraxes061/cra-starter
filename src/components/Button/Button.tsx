import React, { FunctionComponent } from 'react'
import cx from 'classnames'

import './Button.scss'



type ButtonTheme = 'primary' | 'secondary'
type ButtonSize = 'small' | 'medium' | 'large'

type Props = JSX.IntrinsicElements['button'] & {
    readonly theme?: ButtonTheme
    readonly size?: ButtonSize
    readonly isLoading?: boolean
}
const Button: FunctionComponent<Props> = ({
    theme,
    size = 'medium',
    isLoading,
    children,
    ...rest
}) => {
    const classNames = cx(
        'c-button',
        {
            [`c-button--${theme}`]: theme,
            [`c-button--${size}`]: size,
            isLoading: isLoading,
        }
    )
    return (
        <button className={classNames} {...rest}>
            {isLoading
                ? <span>Loading</span>
                : <span>{children}</span>
            }
        </button>
    )
}

export default Button;