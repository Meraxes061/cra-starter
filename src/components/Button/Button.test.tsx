import React from 'react'
import { shallow, mount } from 'enzyme'
import Button from './Button'

describe('<Button/>', () => {
    it('renders correctly default', () => {
        const wrapper = shallow(<Button>Default Button</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly default with loading', () => {
        const wrapper = shallow(<Button isLoading>Default Button</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly default with size small', () => {
        const wrapper = shallow(<Button size='small'>Default Button Small</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly default with size medium', () => {
        const wrapper = shallow(<Button size='medium'>Default Button Medium</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly default with size large', () => {
        const wrapper = shallow(<Button size='large'>Default Button Large</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly primary', () => {
        const wrapper = shallow(<Button theme='primary'>Primary Button</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly primary with size small', () => {
        const wrapper = shallow(<Button theme='primary' size='small'>Primary Button Small</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly primary with size medium', () => {
        const wrapper = shallow(<Button theme='primary' size='medium'>Primary Button Medium</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly primary with size large', () => {
        const wrapper = shallow(<Button theme='primary' size='large'>Primary Button Large</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly secondary', () => {
        const wrapper = shallow(<Button theme='secondary'>Secondary Button</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly secondary with small', () => {
        const wrapper = shallow(<Button theme='secondary' size='small'>Secondary Button Small</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly secondary with medium', () => {
        const wrapper = shallow(<Button theme='secondary' size='medium'>Secondary Button Medium</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('renders correctly secondary with large', () => {
        const wrapper = shallow(<Button theme='secondary' size='large'>Secondary Button Large</Button>)
        expect(wrapper).toMatchSnapshot()
    })
    it('button onclick work correct', () => {
        const fn = jest.fn()
        const wrapper = mount(<Button onClick={fn}>Default</Button>)
        wrapper.find('button').simulate('click')
        expect(fn).toHaveBeenCalled()
    })

})