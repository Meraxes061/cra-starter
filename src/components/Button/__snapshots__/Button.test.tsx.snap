// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders correctly default 1`] = `
<button
  className="c-button c-button--medium"
>
  <span>
    Default Button
  </span>
</button>
`;

exports[`<Button/> renders correctly default with loading 1`] = `
<button
  className="c-button c-button--medium isLoading"
>
  <span>
    Loading
  </span>
</button>
`;

exports[`<Button/> renders correctly default with size large 1`] = `
<button
  className="c-button c-button--large"
>
  <span>
    Default Button Large
  </span>
</button>
`;

exports[`<Button/> renders correctly default with size medium 1`] = `
<button
  className="c-button c-button--medium"
>
  <span>
    Default Button Medium
  </span>
</button>
`;

exports[`<Button/> renders correctly default with size small 1`] = `
<button
  className="c-button c-button--small"
>
  <span>
    Default Button Small
  </span>
</button>
`;

exports[`<Button/> renders correctly primary 1`] = `
<button
  className="c-button c-button--primary c-button--medium"
>
  <span>
    Primary Button
  </span>
</button>
`;

exports[`<Button/> renders correctly primary with size large 1`] = `
<button
  className="c-button c-button--primary c-button--large"
>
  <span>
    Primary Button Large
  </span>
</button>
`;

exports[`<Button/> renders correctly primary with size medium 1`] = `
<button
  className="c-button c-button--primary c-button--medium"
>
  <span>
    Primary Button Medium
  </span>
</button>
`;

exports[`<Button/> renders correctly primary with size small 1`] = `
<button
  className="c-button c-button--primary c-button--small"
>
  <span>
    Primary Button Small
  </span>
</button>
`;

exports[`<Button/> renders correctly secondary 1`] = `
<button
  className="c-button c-button--secondary c-button--medium"
>
  <span>
    Secondary Button
  </span>
</button>
`;

exports[`<Button/> renders correctly secondary with large 1`] = `
<button
  className="c-button c-button--secondary c-button--large"
>
  <span>
    Secondary Button Large
  </span>
</button>
`;

exports[`<Button/> renders correctly secondary with medium 1`] = `
<button
  className="c-button c-button--secondary c-button--medium"
>
  <span>
    Secondary Button Medium
  </span>
</button>
`;

exports[`<Button/> renders correctly secondary with small 1`] = `
<button
  className="c-button c-button--secondary c-button--small"
>
  <span>
    Secondary Button Small
  </span>
</button>
`;
